SalesforceCommerceCloud/pwa-kit
# [Storefront Preview] - 3P CDN, add search parameters to be returned on context refresh

Added a new prop to storefront preview so that querystring parameters can be returned to the retail react app after context is applied in runtime-admin-ui. 

# Description

I added a new prop `getSearchParamsToAppend` to the StorefrontPreview component. This allows customers to indicate search parameters they need returned to the retail react app after context applied in storefront preview/runtime-admin-ui. 

It currently supports two placeholder tokens, but I would love for us to be able to send more of the current context vars (`customQualifiers`, `customerGroupIds`, `sourceCode`)

1. `{{effectiveDateTime}}` : the storefront preview date time as-is with no updates: `2023-10-10T00:39:00Z`
2. `{{effectiveDateTimeUnix}}`: the storefront preview date as a unix time stamp: `1696898340000`

# Types of Changes




> Breaking changes include:
>
> - Removing a public function or component or prop
> - Adding a required argument to a function
> - Changing the data type of a function parameter or return value
> - Adding a new peer dependency to `package.json`

# Changes

- (change1)

# How to Test-Drive This PR

- You need to test this change in the Preview context
- Check out the code
- Build your pwa-kit-react-sdk
- Apply test params as seen [here](https://github.com/sandragolden/pwa-kit/blob/sgolden/feature/release-2.8-preview-add-search-param/packages/pwa-kit-react-sdk/src/ssr/universal/components/storefront-preview/index.test.js#L73-L84)
- Run the storefront
- Run Runtime Admin
- Go to: http://localhost:4000/kyle-preview-demo/p/test-sandra/preview
- Observe that script is loaded as expected
- Apply context, confirm parameters are present in the redirect 

# Checklists




## General


## Accessibility Compliance

You must check off all items in **one** of the follow two lists:


_or..._


## Localization
